content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,150 @@
+
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    kubectl get cm/cluster-info --namespace kube-public -o yaml | \
+        /bin/sed "s/\(certificate-authority-data:\).*/\1 ${base64_encoded_ca}" | \
+        kubectl apply -f -
+    ```
+
+1. Update all service account tokens with the new CA certificates.
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    for namespace in $(kubectl get ns --no-headers | awk '{print $1}'); do
+        for token in $(kubectl get secrets --namespace "$namespace" --field-selector type=kubernetes.io/service-account-token -o name); do
+            kubectl get $token --namespace "$namespace" -o yaml | \
+            /bin/sed "s/\(ca.crt:\).*/\1 ${base64_encoded_ca}" | \
+            kubectl apply -f -
+        done
+    done
+
+    ```
+
+1. Update certificates for user accounts.
+
+    For information about creating certificates for individual user accounts, see [Configure certificates for user accounts](/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts).
+
+    Update `client-certificate-data` and `client-key-data` respectively with the contents of certificates and private keys generated in their respective conf files, along with the kubelet client file in`/var/lib/kubelet/kubelet-client-current.pem`.
+
+1. Update `kubelet.conf`, the `kubelet` certificate, and the private key on all worker nodes as well.
+
+1. Update the `certificate-authority-data` section in all `admin.conf`, `controller-manager.conf`,  `scheduler.conf`, and `kubelet.conf` files with Base64-encoded CA certificate data:
+
+    Example:
+
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    /bin/sed -i "s/\(certificate-authority-data:\).*/\1 ${base64_encoded_ca}" /path/to/conf/files
+    ```
+
+1. Follow these steps on all control plane and worker nodes.
+
+    1. Stop the kubelet service on all control plane and worker nodes.
+
+    1. Kill the containers manually on control plane nodes:
+
+        Below example assumes the container runtime in use is `Docker`.
+
+        ```shell        
+        docker ps -af --name="k8s_*" -q | xargs --no-run-if-empty docker rm -f -
+        ``` 
+
+    1. Start the kubelet service.
+
+    1. Restart any other *[aggregated api servers](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)* or *webhook handlers* to trust the new CA certificates.
+
+    1. Make sure the control plane comes up and logs no TLS errors.
+
+    1. Make sure container networking functions.
+
+    1. Make sure `coredns` and `kube-proxy` are updated with the new CA and function correctly after CA rotation.
+
+    1. Kill or restart all other pods, one by one for all available namespaces, leveraging the benefit of rolling restarts to refresh service account credentials:
+
+        ```shell
+        kubectl delete pods --namespace <namespace>
```

Comment:
Deleting all pods in a namespace at once is a little heavy handed. Annotating a deployment or a daemonset will trigger new pod creation in a safer fashion.

```bash
for namespace in $(kubectl get namespace -o jsonpath='{.items[*].metadata.name}'); do
  for deployment in $(kubectl get deployments -n $namespace -o jsonpath='{.items[*].metadata.name}'); do
    kubectl patch deployment ${name} -p '{"spec":{"template":{"metadata":{"annotations":{"ca-rotation": "1"}}}}}';
  done
  for daemonset in $(kubectl get daemonset -n $namespace -o jsonpath='{.items[*].metadata.name}'); do
    kubectl patch daemonset ${name} -p '{"spec":{"template":{"metadata":{"annotations":{"ca-rotation": "1"}}}}}';
  done
done
```

Reply:
included, thanks